$MirOS: ports/graphics/lcms/patches/patch-include_lcms_h,v 1.2 2005/04/28 20:24:19 tg Exp $
$OpenBSD: patch-include_lcms_h,v 1.1 2004/02/03 05:15:16 brad Exp $
--- include/lcms.h.orig	Sat Nov 26 14:23:25 2005
+++ include/lcms.h	Wed Jan 11 17:16:47 2006
@@ -133,9 +133,12 @@
 
 #if defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__)
 #  include <sys/types.h>
+#  if BYTE_ORDER == BIG_ENDIAN
+#    define USE_BIG_ENDIAN    1
+#  endif
 #  define USE_INT64           1
-#  define LCMSSLONGLONG       int_64_t
-#  define LCMSULONGLONG       u_int64_t
+#  define LCMSSLONGLONG       int64_t
+#  define LCMSULONGLONG       uint64_t
 #endif
 
 #ifdef USE_INT64
@@ -151,7 +154,7 @@
 
 #include <string.h>
 
-#if defined(__GNUC__) || defined(__FreeBSD__)
+#if defined(__GNUC__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__)
 #   include <unistd.h>
 #endif
 
